CONTRIBUTING.md

```diff
@@ -81,39 +81,58 @@ cherry-picked into `book-v2` branch.
 
-Note that PROJECT version should only be bumped if a breaking change is introduced in the PROJECT file format itself.  Changes to the Go scaffolding or the KubeBuilder CLI *do not* affect the PROJECT version.
+### Incrementing versions
 
-Similarly, the introduction of a new major version (`(x+1).0.0`) of the Go plugin might only lead to a new minor (`x.(y+1).0`) release of KubeBuilder, since no breaking change is being made to the CLI itself.  It'd only be a breaking change to KubeBuilder if we remove support for an older version of the plugin.
+Project versions should only be increased if a breaking change is introduced in the PROJECT file scheme itself. Changes to the Go scaffolding or the KubeBuilder CLI *do not* affect project version.
 
-For more information on how the release and plugin versions work, see the the [semver](https://semver.org/) documentation.
+Similarly, the introduction of a new major version (`v(X+1).Y`) of the Go plugin might only lead to a new minor (`vX.(Y+1).Z`) release of KubeBuilder, since no breaking change is being made to the CLI itself. It'd only be a breaking change to KubeBuilder if we remove support for an older version of the plugin.
 
-**NOTE:** In the case of the `v1` and `v2` PROJECT version, a corresponding Plugin version is implied and constant -- `v1` implies the `go.kubebuilder.io/v1` Plugin, and similarly for `v2`.  This is for legacy purposes -- no such implication is made with the `v3` PROJECT version.
+For more information on how the release and plugin versions work, see the [semver](https://semver.org/) documentation.
 
-## Introducing changes in the scaffold files
+**NOTE:** schema for project versions `"1"` and `"2"` were defined before the concept of plugins was introduced, so plugin `go.kubebuilder.io/v2.0` is implicitly used for those project types. Schema for project versions `"3-alpha"` and beyond define a `layout` key that informs the plugin system of which plugin to use.
 
-Changes in the scaffolded files require a new Plugin version. If we delete or update a file that is scaffolded by default, it's a breaking change and requires a `MAJOR` Plugin version.  If we add a new file, it may not be a breaking change.
+## Introducing changes to plugins
 
-**More simply:** any change that will break the expected behaviour of a project built with the previous `MINOR` Plugin versions is a breaking change to that plugin. 
+Changes made to plugins may require a plugin version increase depending on whether
+they break the current version or not. See the plugins design doc [versioning section][plugins-design-versioning]
+for more details.
 
-**EXAMPLE:**
+Breaking changes require either a new Kubebuilder _minor_ or _major_ release.
+A new _major_ release is only necessary for changes that fundamentally alter
+all `kubebuilder` projects.
```

Comment:
The above is very confusing for me. See: 

`Breaking changes require either a new Kubebuilder _minor_ or _major_ release`

If we are following the semantic version convention it shows make no sense. How a breaking change can require either? What does it means? 

`A new _major_ release is only necessary for changes that fundamentally alter
all `kubebuilder` projects.`

What do you mean with  `kubebuilder` projects.` here? What is a change that fundamentally alter all `kubebuilder` projects.?

Reply:
>What means a change that fundamentally alter all kubebuilder projects.?

I agree this is a little nebulous and could use some refinement. Thoughts @DirectXMan12 @joelanford?

Reply:
The doc defines the proposal solution, but I am not sure if should be used as a reference to define what is not breeding cange and how to contribute with.  I would prefer to keep a simple and no verbose explanation here as before.

Reply:
We have a concrete definition of breaking plugin changes now, and no discussion of releasing kubebuilder in relation to plugins.